### PR TITLE
CCO-580: Upgrade to Kubernetes 1.30

### PR DESCRIPTION
**Pull request by jstuever**
This change upgrades the dependencies to Kubernetes 1.30. This requires a controller-runtime update to be compatible with the newer client-go.

### CODE REVIEW

**Comment by jstuever** [comment]
Depends on https://github.com/openshift/cloud-credential-operator/pull/719

**Comment by dlom** [comment]
@jstuever here's how I got code generation/verify working again in hive: https://github.com/dlom/hive/commit/b9e81e1bd3c77f89e7b875db76f1c7b6e6bfdb96

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go [hunk 1]
```diff
@@ -284,41 +284,82 @@ func add(mgr, adminMgr manager.Manager, r reconcile.Reconciler) error {
...
-		allCredRequestsMapFn,
-		configMapPredicate)
+		source.Kind(operatorCache, &corev1.ConfigMap{},
+			configMapAllCredRequestsMapFn,
+			configMapPredicate))
 	if err != nil {
 		return err
 	}
 
+	// configMapAllCredRequestsMapFn simply looks up all CredentialsRequests and requests they be reconciled.
```
Comment (dlom) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:338]: s/config/credential

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go [hunk 2]
```diff
@@ -284,41 +284,82 @@ func add(mgr, adminMgr manager.Manager, r reconcile.Reconciler) error {
...
+	credentialMapAllCredRequestsMapFn := handler.TypedEnqueueRequestsFromMapFunc(func(ctx context.Context, a *operatorv1.CloudCredential) []reconcile.Request {
```
Comment (dlom) [inline pkg/operator/credentialsrequest/credentialsrequest_controller.go:339]: I think this function could be made generic and combined with the similarly named function on line 295.  Neither function actually references the `a` parameter anyways

Comment (jstuever) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go:339]: They don't reference the parameter, but the parameter is of different type.

Comment (jstuever) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go:339]: I wanted to break this out into its own function, but they use `mgr` inline, which can't really be passed any other way.

File: pkg/operator/credentialsrequest/credentialsrequest_controller.go (hunk 1 above)
Comment (jstuever) [reply pkg/operator/credentialsrequest/credentialsrequest_controller.go]: Good catch! [Done](https://github.com/openshift/cloud-credential-operator/compare/a59a050e97e61d6ec59ae4eaa75ea581dbb93de5..e8ed3b77640f58d32912a590527bc7b105962c08).

**Pull request merged**